docs/system/charge_controller.md

```diff
@@ -1,9 +1,94 @@
+---
+sidebarDepth: 2
+---
+
 # Charge Controller
 
+A charge controller regulates the voltage and/or current flowing into batteries. By doing so, it prevents the batteries from overcharging and ensures good battery lifetime.
+
+## Solar
+
+Solar energy is the predominant method for off-grid electricity generation, as it does not require any moveable parts and is easily installed.
+PWM (Pulse Width Modulation) charge controller is in pratice a switch connected between solar panel and a battery. During bulk or constant current load, the switch is simply closed and the battery is charged with whatever current the panel can provide. The panel's voltage is pulled down which effectively lowers the panels efficiency since it is not operating in its **maximum power point**. After bulk charging, the pwm-controller will apply different duty cycles resulting in current boosts, effectively applying a constant voltage. The cycles are adapted depending on the the battery's charge state. The **interactive** example in Figure [1] shows the power effectively used for charging a battery with a given base voltage. The closer the rated battery voltage to the maximum power point voltage, the higher the overall efficiency.
+
+As explained in the [Solar Panel](solar_panel.md) chapter, the voltage of a solar panel depends on the number of cells, the temperature, the irradiance and the amount current draw.
+
+Most 12 V solar panels output 16-20 V in their maximum power point, whereas 12 V batteries usually need around 14-14.5 V to get fully charged. Direct connection of batteries to the solar panel without any regulation causes damage to the batteries as a result of overcharging.
```

Comment:
```suggestion
Most 12 V solar panels output 16-20 V in their maximum power point, whereas 12 V batteries usually need around 14-14.5 V to get fully charged. Continuous connection of batteries to the solar panel without any regulation would cause damage to the batteries as a result of overcharging.
```